inference-engine/src/auto_plugin/auto_infer_request.cpp

```diff
@@ -11,32 +12,41 @@ namespace AutoPlugin {
 
 AutoInferRequest::AutoInferRequest(const InputsDataMap&              networkInputs,
                                    const OutputsDataMap&             networkOutputs,
-                                   const SoIInferRequestInternal&    inferRequest)
+                                   const SoIInferRequestInternal&    inferRequest,
+                                   AutoPlugin::NetworkSharedFuture f, std::atomic<bool>& anyRequestHasHotSwapped)
     : IInferRequestInternal(networkInputs, networkOutputs)
-    , _inferRequest(inferRequest) {
+    , _inferRequest(inferRequest), _sharedFutureForActualNetwork(f), _anyRequestHasHotSwapped(anyRequestHasHotSwapped) {
+    for (const auto &it : _networkInputs)
+        _inputs[it.first] = _inferRequest->GetBlob(it.first);
+    for (const auto &it : _networkOutputs)
+        _outputs[it.first] = _inferRequest->GetBlob(it.first);
 }
 
 std::map<std::string, InferenceEngine::InferenceEngineProfileInfo> AutoInferRequest::GetPerformanceCounts() const {
     return _inferRequest->GetPerformanceCounts();
 }
 
 void AutoInferRequest::InferImpl() {
+    HotSwapRequests(); //safe to call here (before actual inference started)
+    SetBlobsToDeviceRequest();
     _inferRequest->Infer();
 }
 
 void AutoInferRequest::SetBlob(const std::string& name, const InferenceEngine::Blob::Ptr& data) {
-    _inferRequest->SetBlob(name, data);
+    IInferRequestInternal::SetBlob(name, data);
 }
 
 Blob::Ptr AutoInferRequest::GetBlob(const std::string& name) {
-    return _inferRequest->GetBlob(name);
+    return IInferRequestInternal::GetBlob(name);
 }
 
 void AutoInferRequest::Cancel() {
     _inferRequest->Cancel();
 }
 
 void AutoInferRequest::StartAsync() {
+    HotSwapRequests(); //safe to call here (before actual inference started)
```

Comment:
this is a caveat on async: as calling the STartAsync from second therad should throw (REQUEST_BUSY, afaik), we need to make sure that another StartAsync still would throw even though the HotSwap has happend in the "current" StartAsync.

Maybe creating a special callback (that alos wraps original user's callback is better idea).
Or, inheriting the AutoRequest from the InferRequestThreadSafeDefault and checking it's state (as opposite to relying on the BUSY state of the device's request) is also possible